src/charm.py

```diff
@@ -138,6 +138,7 @@ def _on_config_changed(self, event):
             - Writing config to config files\
             - Restarting zookeeper service
         """
+        self.snap.refresh(self.config["snap-channel"])
```

Comment:
Suggestion to remove this for the time being, until we have multiple snap channels to test behavior, and that the service doesn't blow up. I gather from the snap team that there's a lot of clever config we can do for the snap to handle this (`refresh-mode` and `reload-command`), but that's out of scope for this work imo. Just noting here for posterity.

As such, can we add a check here to say `if snap-channel different: BlockedStatus("cannot set at runtime")`, or something, a useful way of letting the user know that their change to snap-channel didn't go through but the service is still running.